addons/l10n_mx/controllers/portal.py

```diff
@@ -0,0 +1,31 @@
+from itertools import count
+
+from odoo.addons.portal.controllers import portal
+from odoo.http import request
+
+class CustomerPortal(portal.CustomerPortal):
+
+    def _get_mandatory_fields(self):
+
+        country_id = request.env.context.get('portal_form_country_id')
+        if country_id:
+            try:
+                country_id = int(country_id)
+            except ValueError:
+                country_id = None
+        if country_id and request.env['res.country'].sudo().browse(country_id).code == 'MX':
+            return super()._get_mandatory_fields() + ['zipcode', 'vat']
+        return super()._get_mandatory_fields()
+
+    def _get_optional_fields(self):
```

Comment:
(all previous comments apply on this method too) :wink:

Reply:
@yosuanicolaus  Thank you for the feedback! The changes have been pushed